### PR TITLE
fix: warn_unused_result

### PR DESCRIPTION
**Pull request by sangshuduo**
gcc 12.2.0 on Ubuntu 23.04

annoying if -Werror enabled

### CODE REVIEW

**Comment by sangshuduo** [comment]
Would anybody take care of this PR? Any comment or advice? Thanks.

File: src/unix/linux.c [hunk 1]
```diff
@@ -1681,7 +1681,10 @@ int uv_cpu_info(uv_cpu_info_t** ci, int* count) {
     return UV__ERR(errno);
   }
 
-  fgets(buf, sizeof(buf), fp);  /* Skip first line. */
+  if (fgets(buf, sizeof(buf), fp) == NULL) {
+    fclose(fp);
+    return UV_EINVAL;
```
Comment (trevnorris) [inline src/unix/linux.c:1686]: For consistency with the rest of the function, I'd assign `fp = NULL;` after the call to `fclose()`.

**Review by trevnorris — COMMENTED** [review]
I'm fine with the change. Just left one comment.

@bnoordhuis you have any feedback?

File: src/unix/linux.c (hunk 1 above)
Comment (sangshuduo) [reply src/unix/linux.c:1724]: Thanks for your comments. I resolved it with a new commit.

Comment (bnoordhuis) [reply src/unix/linux.c:1724]: There's a memory leak now, the `cpus` array isn't freed. Also applies to all the other early returns.

That aside, we're reading from a /proc pseudo-file where reads can't fail so this is all pointless ceremony to appease an overzealous compiler. If you really want to get rid of the warnings, change the fread() calls to this:
```c
if (NULL == fgets(...))
  abort();
```
(and note the yoda style, I'm particular about that)

Comment (sangshuduo) [reply src/unix/linux.c]: I fixed it with a new commit. Just learn the style has a name. :)

File: src/unix/linux.c [hunk 2]
```diff
@@ -1718,7 +1718,9 @@ int uv_cpu_info(uv_cpu_info_t** ci, int* count) {
     return UV__ERR(errno);
   }
 
-  fgets(buf, sizeof(buf), fp);  /* Skip first line. */
+  if (NULL == fgets(buf, sizeof(buf), fp)) {
+    abort();
+  }  /* Skip first line. */
```
Comment (bnoordhuis) [inline src/unix/linux.c:1723]:
```suggestion
  if (NULL == fgets(buf, sizeof(buf), fp))
    abort();
```

File: src/unix/linux.c [hunk 3]
```diff
@@ -1729,7 +1731,9 @@ int uv_cpu_info(uv_cpu_info_t** ci, int* count) {
     if (n != 7)
       break;
 
-    fgets(buf, sizeof(buf), fp);  /* Skip rest of line. */
+    if (NULL == fgets(buf, sizeof(buf), fp)) {
+      abort();
+    }  /* Skip rest of line. */
```
Comment (bnoordhuis) [inline src/unix/linux.c:1736]:
```suggestion
    if (NULL == fgets(buf, sizeof(buf), fp))
      abort();
```

File: src/unix/linux.c [hunk 4]
```diff
@@ -1809,7 +1813,11 @@ int uv_cpu_info(uv_cpu_info_t** ci, int* count) {
     if (fp == NULL)
       continue;
 
-    fscanf(fp, "%llu", &(*cpus)[cpu].freq);
+    if (EOF == fscanf(fp, "%llu", &(*cpus)[cpu].freq)) {
+      fclose(fp);
+      fp = NULL;
+      return UV_EINVAL;
+    }
```
Comment (bnoordhuis) [inline src/unix/linux.c:1820]:
```suggestion
    if (1 != fscanf(fp, "%llu", &(*cpus)[cpu].freq))
      abort();
```

**Review by bnoordhuis — APPROVED** [review]

**Pull request merged**